wandb/sdk/launch/sweeps/scheduler.py

```diff
@@ -362,15 +358,22 @@ def exit(self) -> None:
                 f"{LOG_PREFIX}Failed to save state: {traceback.format_exc()}"
             )
 
-        if self.state not in [
-            SchedulerState.COMPLETED,
-            SchedulerState.STOPPED,
-        ]:
+        status = ""
+        if self.state == SchedulerState.FLUSH_RUNS:
+            self._set_sweep_state("PAUSED")
+            status = "paused"
+        elif self.state == SchedulerState.COMPLETED:
+            self._set_sweep_state("FINISHED")
+            status = "completed"
+        elif self.state in [SchedulerState.CANCELLED, SchedulerState.STOPPED]:
+            self._set_sweep_state("CANCELED")  # one L
+            status = "cancelled"
+        else:
             self.state = SchedulerState.FAILED
             self._set_sweep_state("CRASHED")
-        else:
-            self._set_sweep_state("FINISHED")
+            status = "crashed"
 
+        wandb.termlog(f"{LOG_PREFIX}Scheduler {status}")
         self._stop_runs()
```

Comment:
In the paused state we shouldn't stop the runs no?

Reply:
Same with finished?

Reply:
good point, stop_runs doesn't do anything when paused or finished, because to get in that state the runs must be completed (we wait for all runs to complete). ill make that more obvious